pkg/scheduler/reconciliation/reconciliation_test.go

```diff
@@ -729,3 +681,131 @@ func dbConnection(t *testing.T) db.Connection {
 	}
 	return dbConn
 }
+
+func TestReconciliationParallel(t *testing.T) {
+
+	t.Run("Create multiple instances of single reconciliations in parallel", func(t *testing.T) {
+		//initialize WaitGroup
+		var wg sync.WaitGroup
+		//reset db connection
+		dbConn = nil
+
+		repo := newPersistentRepository(t)
+		inventory, err := cluster.NewInventory(db.NewTestConnection(t), true, cluster.MetricsCollectorMock{})
+		require.NoError(t, err)
+
+		errChannel := make(chan error, 100)
+		mockClusterState, _ := createClusterStates(t, inventory)
+
+		defer func() {
+			require.NoError(t, inventory.Delete(mockClusterState.Cluster.RuntimeID))
+		}()
+
+		startAt := time.Now().Add(1 * time.Second)
+		for i := 0; i < 50; i++ {
+			wg.Add(1)
+			go func(errChannel chan error, repo Repository) {
+				defer wg.Done()
+				time.Sleep(time.Until(startAt))
+				_, err := repo.CreateReconciliation(mockClusterState, nil)
+				if err != nil {
+					errChannel <- err
+				}
+			}(errChannel, repo)
+		}
+		wg.Wait()
+		recons, err := repo.GetReconciliations(nil)
+		require.NoError(t, err)
+		require.Equal(t, 1, len(recons))
+		require.False(t, recons[0].Finished)
+		require.Equal(t, 49, len(errChannel))
+	})
+
+	t.Run("Update single operation state in multiple parallel threads", func(t *testing.T) {
+		//initialize WaitGroup
+		var wg sync.WaitGroup
+		//reset db connection
+		dbConn = nil
+
+		repo := newPersistentRepository(t)
+		inventory, err := cluster.NewInventory(db.NewTestConnection(t), true, cluster.MetricsCollectorMock{})
+		require.NoError(t, err)
+
+		errChannel := make(chan error, 100)
```

Comment:
See above.. would recommend to use a variable for the length and for the loop to show that both belong together